health-checker/src/main/java/io/stargate/health/CPUGaugeMetricSet.java

```diff
@@ -0,0 +1,97 @@
+package io.stargate.health;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import oshi.SystemInfo;
+import oshi.hardware.CentralProcessor;
+import oshi.hardware.CentralProcessor.TickType;
+
+public class CPUGaugeMetricSet implements MetricSet {
+
+  final CentralProcessor processor;
+  private long tickTime;
+  private final long[] prevTicks;
+  private final long[] curTicks;
+
+  public CPUGaugeMetricSet() {
+    processor = new SystemInfo().getHardware().getProcessor();
+    this.prevTicks = new long[TickType.values().length];
+    this.curTicks = new long[TickType.values().length];
+  }
+
+  @Override
+  public Map<String, Metric> getMetrics() {
+    final Map<String, Metric> gauges = new HashMap<>();
+
+    gauges.put("load", (Gauge<Double>) this::getSystemCpuLoad);
+    gauges.put("user", (Gauge<Double>) () -> getTick("user"));
+    gauges.put("nice", (Gauge<Double>) () -> getTick("nice"));
+    gauges.put("sys", (Gauge<Double>) () -> getTick("sys"));
+    gauges.put("idle", (Gauge<Double>) () -> getTick("idle"));
+    gauges.put("iowait", (Gauge<Double>) () -> getTick("iowait"));
+
+    return Collections.unmodifiableMap(gauges);
+  }
+
+  private Double getSystemCpuLoad() {
+    long now = System.currentTimeMillis();
+    if (now - this.tickTime > 950) {
+      // Enough time has elapsed.
+      updateSystemTicks();
+    }
+
+    return processor.getSystemCpuLoadBetweenTicks(prevTicks);
+  }
+
+  private Double getTick(String tickType) {
+    long now = System.currentTimeMillis();
+    if (now - this.tickTime > 950) {
+      // Enough time has elapsed.
+      updateSystemTicks();
+    }
+
+    long user = curTicks[TickType.USER.getIndex()] - prevTicks[TickType.USER.getIndex()];
+    long nice = curTicks[TickType.NICE.getIndex()] - prevTicks[TickType.NICE.getIndex()];
+    long sys = curTicks[TickType.SYSTEM.getIndex()] - prevTicks[TickType.SYSTEM.getIndex()];
+    long idle = curTicks[TickType.IDLE.getIndex()] - prevTicks[TickType.IDLE.getIndex()];
+    long iowait = curTicks[TickType.IOWAIT.getIndex()] - prevTicks[TickType.IOWAIT.getIndex()];
+    long totalCPU = user + nice + sys + idle + iowait;
```

Comment:
I do not really know how `oshi.hardware` reports data, but is it guaranteed that these numbers properly add up to the "tocal CPU"? I'd guess total should be a deterministic value based on the CPU frequency and time interval... is there a method in this tool to get the total without adding up components?

Reply:
> is there a method in this tool to get the total without adding up components

No, unfortunately you just have and array of "ticks"

Reply:
(also mention is another comment, but relevant here)
This example uses more ticks for calculating `totalCPU`:
https://github.com/goodyinxin/eladmin/blob/4adea4ebe2c4ca6f1c691f64036cc96a4a10737b/eladmin-monitor/src/test/java/me/zhengjie/SystemInfoTest.java#L194-L202

If the tool reports CPU usage as a collection of ticks, I suppose we ought to use them all for `totalCPU` even if we do not expose some of them as metrics.